appcheck/appcheck.go

```diff
@@ -83,8 +92,9 @@ func NewClient(ctx context.Context, conf *internal.AppCheckConfig) (*Client, err
 	}
 
 	return &Client{
-		projectID: conf.ProjectID,
-		jwks:      jwks,
+		projectID:        conf.ProjectID,
+		jwks:             jwks,
+		tokenVerifierUrl: buildTokenVerifierUrl(conf.ProjectID),
```

Comment:
Instead of passing the URL you can pass the http client here and build the URL in `VerifyOneTimeToken` (see the comment below on that)
```go
hc, _, err := internal.NewHTTPClient(ctx, conf.Opts...)
if err != nil {
	return nil, err
}

return &Client{
	projectID:  conf.ProjectID,
	jwks:       jwks,
	httpClient: hc,
}, nil
```

Reply:
For this to work correctly you will also need to update the following:

`AppCheckConfig struct` in `firebase-admin-go/internal/internal.go`
```go
type AppCheckConfig struct {
	Opts      []option.ClientOption // <----------------
	ProjectID string
}
```

`func (a *App) AppCheck(ctx context.Context) (*appcheck.Client, error) {` in `firebase-admin-go/firebase.go`
```go
func (a *App) AppCheck(ctx context.Context) (*appcheck.Client, error) {
	conf := &internal.AppCheckConfig{
		ProjectID: a.projectID,
		Opts:      a.opts, // <-------------------
	}
	return appcheck.NewClient(ctx, conf)
}
```